crates/hir-expand/src/fixup.rs

```diff
@@ -476,12 +465,11 @@ fn reverse_fixups_(tt: &mut TopSubtree, undo_info: &[TopSubtree]) {
             }
         }
         tt::TokenTree::Subtree(tt) => {
+            // If only one of the delimiters is fake, we still want to keep the subtree.
+            // (FIXME: Should we actually just 'unwrap' the subtree?)
             if tt.delimiter.close.anchor.ast_id == FIXUP_DUMMY_AST_ID
-                || tt.delimiter.open.anchor.ast_id == FIXUP_DUMMY_AST_ID
+                && tt.delimiter.open.anchor.ast_id == FIXUP_DUMMY_AST_ID
```

Comment:
That is because the syntax bridge will remove these elements right? Might be good to add to the comment since we should never leak the `FIXUP_DUMMY_AST_ID`s

Reply:
Hmm no, it was because I made the ArgList fixup add the closing parenthesis in the `foo(a` case. I changed it back to `||`, so we instead rely on `convert_tokens` to keep trees balanced.